### ISSUE TITLE
Activity diagram leaving out [else]

### ISSUE THREAD

**Issue by gerhean** [nus-cs2103-AY1920S2/forum#112]
Is it ok to leave out the [else] on alternate paths? This is especially for long else if chains which make the diagram feel fairly cluttered. So that i can use the if else in plantuml as shown here.

![image](https://user-images.githubusercontent.com/16630400/79035191-aa6e2700-7bee-11ea-8f27-c996177f6e8b.png)

**Comment by okkhoy** [comment]
Isn't your non-yes path the else path for each of the conditionals? 

You can choose to have one merge node and add a UML note to indicate it collects all the branches.

BTW, putting the conditional inside the hexagon is not standard UML. You need to use a diamond and put the true case as one guard clause and the false case as the other branch.

**Comment by eugenetyc** [comment]
Hi Prof, I have not managed to find a way to have more than 2 out-branches from a branch node, without either:

Use PlantUML Beta and do what Ger Hean is currently doing.

OR

Use the older version of PlantUML and be unable to use [] square brackets to contain the condition.

Would it be okay to do the latter (ie leave out the square brackets) for 3 of more branches?

**Comment by okkhoy** [comment]
2 out meaning the `if` and one `else`? or multiple alternate paths (something like modeling switch statement?) 
Does this help?

```
@startuml
start
 :start;

 if () then ([c1])
   if () then ([c2])
     if () then ([c3])
       : decision;
       stop
       elseif() then ([c32])
       : blah;
     else ([else])
     endif
   else ([else])
   endif
 else ([else])
 endif
 : decision 1;
 stop
 @enduml
```

Gives: 

![image](https://user-images.githubusercontent.com/1023494/79048059-0ff10080-7c4d-11ea-914f-bd0ffe91e454.png)


Messed and mashed the example found originally here: https://stackoverflow.com/questions/56651531/how-to-make-vertical-switch-case-like-this

**Comment by eugenetyc** [comment]
Yes, I mean to say a branch node with multiple branches coming out, like an if, elseif, elseif, elseif, else (5 branches here).

I have not found a way to have 5 branches appear out from a single branch node. Would the method you just showed be acceptable, if what I wanted to portray was a block of code with 5 branches (i.e. if, elseif, elseif, elseif, else)?

**Comment by PotatoCombat** [comment]
Hi @eugenetyc! Did you mean having a diagram that looks like this?

![SortParserActivityDiagram](https://user-images.githubusercontent.com/39809238/79051435-cf9c7d00-7c62-11ea-9b96-16177f75ecbd.png)

If so, I would also like to clarify if the above diagram is *acceptable*.

The example given by @okkhoy has merge nodes at the end, with all the alternate paths represented vertically.

In contrast, our diagram is *horizontal* and does not have the *ending merge node*, because of the way PlantUML creates `if`-`else if`-`else` branches.

**Comment by PotatoCombat** [comment]
Since we are discussing Activity Diagrams, I would like to add on to @gerhean's question.
Do let me know if this should be a separate issue.

I have a few diagrams which have branches like `condition B` in @gerhean's diagram.

I believe what I really need is a way to have **branches join each other**, like the one in the textbook:
<img width="484" alt="Screenshot 2020-04-12 at 2 16 27 AM" src="https://user-images.githubusercontent.com/39809238/79051605-9fa1a980-7c63-11ea-8e8e-3c6a5938aeee.png">

However I was *unable* to find a solution using the **BETA version of PlantUML**.

Is it **acceptable** to have **multiple end nodes** in activity diagrams?

At the moment my diagrams look like this in the DG, along with a note explaining that all the end nodes should actually meet at a single merge node before it.

![SyntaxHighlightingActivityDiagram2](https://user-images.githubusercontent.com/39809238/79051348-484f0980-7c62-11ea-9582-6ba89b14c448.png)

![SyntaxHighlightingActivityDiagram3](https://user-images.githubusercontent.com/39809238/79051350-4b49fa00-7c62-11ea-9417-e3149a456dad.png)

**Comment by okkhoy** [comment]
@eugenetyc 

> I have not found a way to have 5 branches appear out from a single branch node

I think proper UML notation doesn't allow that. It is more of a convenience thing when we hand-draw it (at least in PUML, I couldn't get it to work that way; no time to go back to UML specs to confirm 😞).


@PotatoCombat I used Plant UML as well 😜  
Yes, I agree that the default way PUML generates if-else is horizontal. 
You need to play around to break the `if ... elseif ... endif` to multiple `if ... endif` carefully to get proper merge nodes.

After meddling around for a few minutes trying to get it work by myself, I resorted to SO, (check the source is given).

The standard UML requires every branch to be merged. I think PUML relaxed it to make the diagram easy to understand (again, no time to go back to UML specs to confirm 😞). 
BTW, your diagram is missing the `[else]` guard clause.

> However I was unable to find a solution using the BETA version of PlantUML.

Check the sample code I have given; where you put the `stop` matters. If you don't have the stop, it joins at the corresponding merge node.

**Comment by eugenetyc** [comment]
> Hi @eugenetyc! Did you mean having a diagram that looks like this?
> 
> ![SortParserActivityDiagram](https://user-images.githubusercontent.com/39809238/79051435-cf9c7d00-7c62-11ea-9b96-16177f75ecbd.png)
> 
> If so, I would also like to clarify if the above diagram is _acceptable_.
> 
> The example given by @okkhoy has merge nodes at the end, with all the alternate paths represented vertically.
> 
> In contrast, our diagram is _horizontal_ and does not have the _ending merge node_, because of the way PlantUML creates `if`-`else if`-`else` branches.

Yeah, but without so many branch nodes; just have 1 branch node with multiple branches coming out.

Anyway, it's fine. I just used 2 branch nodes and abstracted the ideas using 1 action rather than display them as many branches.
